scripts/config_osm_data.py

```diff
@@ -484,3 +484,241 @@
+        "MY-SG-BN": "malaysia-singapore-brunei",  # Geofk shortcurt
+        "MV": "maldives",
+        "MN": "mongolia",
+        "MM": "myanmar",
+        "NP": "nepal",
+        "PK": "pakistan",
+        "PH": "philippines",
+        "LK": "sri-lanka",
+        "SY": "syria",
+        "TW": "taiwan",
+        "TJ": "tajikistan",
+        "TH": "thailand",
+        "TM": "turkmenistan",
+        "UZ": "uzbekistan",
+        "VN": "vietnam",
+        "YE": "yemen",
+    },
+    "australia-oceania": {
+        # "AS": "american-oceania", # Islands
+        "AU": "australia",
+        # "CK": 'cook islands',
+        "FJ": "fiji",
+        # "PF": "french-polynesia", # Islands
+        # "GU": "guam", # Island
+        # "KI": "kiribati", # Islands
+        # "MH": 'marshall islands', # Islands
+        # "FM": "micronesia", # Islands
+        # "NR": "nauru", # Islands
+        "NC": "new-caledonia",
+        "NZ": "new-zealand",
+        # "NU": "niue", # Island
+        # "NF": "norfolk island", # Island
+        # "MP": 'northern mariana islands', # Islands
+        # "PW": "palau", # Islands
+        "PG": "papua-new-guinea",
+        # "WS": "samoa", # Islands
+        # 'SB': 'solomon islands', # Islands
+        # "TK": "tokelau", # Islands
+        # "TO": "tonga", # Islands
+        # "TV": "tuvalu", # Islands
+        # "VU": "vanuatu", # Islands
+        # "WF": "wallis-et-futuna", # Islands
+    },
+    "europe": {
+        "AL": "albania",
+        "AD": "andorra",
+        "AT": "austria",
+        "BY": "belarus",
+        "BE": "belgium",
+        "BA": "bosnia-herzegovina",
+        "BG": "bulgaria",
+        "HR": "croatia",
+        "CZ": "czech-republic",
+        "CY": "cyprus",
+        "DK": "denmark",
+        "EE": "estonia",
+        # "FO": "faroe islands", # Islands
+        "FI": "finland",
+        "FR": "france",
+        "GE": "georgia",
+        "DE": "germany",
+        # 'GI': 'gibraltar', # Peninsula; Isolated PS?
+        "GR": "greece",
+        # 'GG': 'guernsey', # Island
+        "HU": "hungary",
+        "IS": "iceland",
+        "IE": "ireland-and-northern-ireland",
+        # 'IM': 'isle of man', # Island
+        "IT": "italy",
+        # 'JE': 'jersey', # Island
+        "LV": "latvia",
+        "LI": "liechtenstein",
+        "LT": "lithuania",
+        "LU": "luxembourg",
+        "MK": "macedonia",
+        "MT": "malta",
+        "MD": "moldova",
+        "MC": "monaco",
+        "ME": "montenegro",
+        "NL": "netherlands",
+        "NO": "norway",
+        "PL": "poland",
+        "PT": "portugal",
+        "RO": "romania",
+        # 'SM': 'san-marino',
+        "RS": "serbia",
+        "SK": "slovakia",
+        "SI": "slovenia",
+        "ES": "spain",
+        # 'SJ': 'svalbard-and-jan-mayen', # Islands
+        "SE": "sweden",
+        "CH": "switzerland",
+        "UA": "ukraine",
+        "GB": "great-britain",
+        "TR": "turkey",
+    },
+    "russia": {
```

Comment:
I see the problem with russia.
Russia is the only nation that is classified as a continent basically, thus the typical downloading format we use does not work.
https://github.com/pypsa-meets-africa/pypsa-africa/blob/8d6a3adc79ab37da00bdfb13ddb67247ea157ff6/scripts/download_osm_data.py#L80-L83

Probably, we may hard code a change here so that if the continent is "russia" and the requested country_name code is also russia then we use the link `f"https://download.geofabrik.de/{continent}-latest.osm.pbf"` instead
Actually, the above concept may be generalized to whenever continent==country_name but may need tests; the hard coding of russia solve that special case

We discussed with @pz-max and he seemed prone to this solution, he proposed the hard fix for russia. Personally, I would like a generalization but looking at geofk it is really the only case...

Reply:
@davide-f, @pz-max many thanks for suggestions. Yes, exactly. The condition `continent ==country_name` in Geofabrik seems to hold for Russia only.

To make things more complicated, it could be more convenient having an opportunity to extract the data bunch related to a single district according to Geofabrik logic. E.g. defining `"russia"` as a `{continent}` and `"far-eastern-fed-district"` as a `{country}`. But hard fix `https://download.geofabrik.de/russia` looks much more elegant.

Reply:
Ok! Hard fix is fine :)